apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/index.tsx

```diff
@@ -230,6 +221,24 @@ export default function DocumentEditor({
   )
 
   const isMerged = commit.mergedAt !== null
+
+  const [forceReadMetadata, setForceReadMetadata] = useState(false)
+  useEffect(() => {
+    if (!forceReadMetadata) return
+
+    async function readMetadata() {
+      await runReadMetadata({
+        prompt: value,
+        documents: _documents,
+        document,
+        fullPath: document.path,
+      })
+      setForceReadMetadata(false)
+    }
+
+    readMetadata()
+  }, [forceReadMetadata, setForceReadMetadata])
```

Comment:
dangerous to have a use effect that modifies a state in which it depends, why can we make this use effect react to whatever it needs to react instead of having this imperative handling

Reply:
Because what I need to react to is inputs. And inputs depends on metadata to produce `parameters` for the AI

If you put in that dependency array `inputs` it start spinning like crazy calls to the worker

Reply:
hmm so let's see if i understand. Metadata read generates inputs which you change when i select a document log which should in turn trigger another metadata read?